gitpod-io/browser-extension
# Default domain restriction

Do not request all domains by default, let the user enable the integration everywhere. 

- The default origins, which are required for the extension to work, are:
	- `https://github.com`
	- `https://gitlab.com`
	- `https://bitbucket.org`
	- `https://gitpod.io`
	- `https://*.gitpod.dev`
	- `https://*.gitpod.cloud`
- This also updates the permissions to include `scripting`, `contextMenus` and `activeTab`, but none of these require user prompts.
- Additionally, updating the Gitpod URL requests the corresponding hostname to be allowed as well.
- Support for the automatic detection of Gitpod Dedicated has been removed, because it did not make sense to keep with the permission changes


## How to test

- Remove the extension if you've had it before (necessary for permissions reset) and update it with a newly built one
- Visit our [Bitbucket Datacenter instance](https://bitbucket.gitpod-dev.com/users/svenefftinge/repos/browser-extension-test/browse) and see the extension does not inject any buttons
- Open the context menu of the Gitpod extension and enable it on the domain. If required, refresh the tab and see the button appear.
